octue/mixins/cool_nameable.py

```diff
@@ -0,0 +1,8 @@
+from coolname import generate_slug
+
+
+class CoolNameable:
+    """ A mixin that gives the subclass a highly unique human-readable name e.g. "melodic-kestrel". """
+
+    def __init__(self):
+        self.cool_name = generate_slug(2)
```

Comment:
I'd anticipated that this would be on the `name` attribute, and that it'd only be generated if the `name` attribute hadn't been set by the superclass initialisation, something like:
```
name = kwargs.pop('name', None):
if name:
    self.name = name
    self._cool_named = false
else:
    self.name = generate_slug(2)
    self._cool_named = true
```

Also, not chaining the initialisation parameters or initialising the superclass in the way that's needed for this to comply with the *able mixin pattern.

Reply:
I think I could use an intro to dealing `__init__` methods from lots of superclasses - I'll have a look into this online or maybe we could have a quick chat about it?

Reply:
@thclark I've addressed these issues although we should note that `CoolNameable` won't currently work with any class that has a `name` property unless we set the name on e.g. `self._name`